server/tracedb/connection/options.go

```diff
@@ -0,0 +1,37 @@
+package connection
+
+import "context"
+
+func WithConnectivityTest(step TestStep) TesterOption {
```

Comment:
This pattern is very interesting, do you have documentation around it? I'd like to understand it better 😄

Reply:
It's called _option pattern_. You can read about it here: https://golang.cafe/blog/golang-functional-options-pattern.html